domain/machine/service/service.go

```diff
@@ -70,3 +83,33 @@ func (s *Service) GetMachineLife(ctx context.Context, machineId string) (*life.L
 	life, err := s.st.GetMachineLife(ctx, machineId)
 	return life, errors.Annotatef(err, "getting life status for machine %q", machineId)
 }
+
+// ListAllMachines returns the ids of all machines in the model.
+func (s *Service) ListAllMachines(ctx context.Context) ([]string, error) {
+	machines, err := s.st.ListAllMachines(ctx)
+	if err != nil {
+		return nil, errors.Annotate(err, "retrieving all machines")
+	}
+	return machines, nil
+}
+
+// InstanceId returns the provider specific instance id for this machine.
+// If the machine is not provisioned, it returns a NotProvisionedError.
+func (s *Service) InstanceId(ctx context.Context, machineId string) (string, error) {
```

Comment:
The wording here is confusing for me. Should we consider settling on either provider or cloud? i.e the func docs and error use to different words to describe the same thing.